Exception is thrown when clicking the "Couch" (Floating Button for "Application State")

Caused by: java.lang.NullPointerException: null
	at org.openbase.bco.bcozy.view.FloatingPopUp.lambda$addElement$3(FloatingPopUp.java:74)
	at org.openbase.bco.bcozy.view.FloatingPopUp.lambda$addChildren$4(FloatingPopUp.java:103)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Node.fireEvent(Node.java:8413)
	at javafx.scene.control.Button.fire(Button.java:185)
	at com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:182)
	at com.sun.javafx.scene.control.skin.BehaviorSkinBase$1.handle(BehaviorSkinBase.java:96)
	at com.sun.javafx.scene.control.skin.BehaviorSkinBase$1.handle(BehaviorSkinBase.java:89)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3757)
	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485)
	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:381)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$354(GlassViewEventHandler.java:417)
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:416)
	at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
	at com.sun.glass.ui.View.notifyMouse(View.java:937)
	... 3 common frames omitted

--- a/src/main/java/org/openbase/bco/bcozy/view/CenterPane.java
+++ b/src/main/java/org/openbase/bco/bcozy/view/CenterPane.java
@@ -65,12 +65,6 @@ public class CenterPane extends StackPane {
 
         appStateProperty = new SimpleObjectProperty<>(CenterPaneController.State.SETTINGS);
         
-        appStateProperty.addListener(new ChangeListener<CenterPaneController.State> () {
-            @Override
-            public void changed(ObservableValue<? extends CenterPaneController.State> observable, CenterPaneController.State oldValue, CenterPaneController.State newValue) {
-                System.out.println(newValue);
-            }
-        });
         // Initializing components
         this.settingsMenu = loadSettingsMenu(height);
 

--- a/src/main/java/org/openbase/bco/bcozy/view/CenterPane.java
+++ b/src/main/java/org/openbase/bco/bcozy/view/CenterPane.java
@@ -35,6 +35,8 @@ import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.net.URL;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
 import org.openbase.jul.exception.CouldNotPerformException;
 import org.openbase.jul.exception.NotAvailableException;
 import org.openbase.jul.exception.printer.ExceptionPrinter;
@@ -62,15 +64,26 @@ public class CenterPane extends StackPane {
     public CenterPane(final double height) {
 
         appStateProperty = new SimpleObjectProperty<>(CenterPaneController.State.SETTINGS);
-
+        
+        appStateProperty.addListener(new ChangeListener<CenterPaneController.State> () {
+            @Override
+            public void changed(ObservableValue<? extends CenterPaneController.State> observable, CenterPaneController.State oldValue, CenterPaneController.State newValue) {
+                System.out.println(newValue);
+            }
+        });
         // Initializing components
         this.settingsMenu = loadSettingsMenu(height);
 
         final FloatingPopUp viewModes = new FloatingPopUp(Pos.BOTTOM_RIGHT);
-        viewModes.addParentElement(MaterialIcon.WEEKEND, (Runnable) null); //TODO: Add EventHandler when needed
-        viewModes.addElement(MaterialDesignIcon.THERMOMETER_LINES, (Runnable) null);//TODO: Add EventHandler when needed
-        viewModes.addElement(MaterialIcon.VISIBILITY, (Runnable) null);//TODO: Add EventHandler when needed
-
+        viewModes.addParentElement(MaterialIcon.WEEKEND, () -> {
+            appStateProperty.set(CenterPaneController.State.MOVEMENT);
+        }); 
+        viewModes.addElement(MaterialDesignIcon.THERMOMETER_LINES, () -> {
+            appStateProperty.set(CenterPaneController.State.TEMPERATURE);
+        }); 
+        viewModes.addElement(MaterialIcon.VISIBILITY, () -> {
+            appStateProperty.set(CenterPaneController.State.SETTINGS);
+        }); 
         final FloatingButton settingsBtn = new FloatingButton(new SVGIcon(MaterialDesignIcon.SETTINGS, Constants.MIDDLE_ICON, true));
 
         this.setAlignment(settingsBtn, Pos.TOP_RIGHT);
@@ -86,6 +99,7 @@ public class CenterPane extends StackPane {
         this.setMinHeight(height);
         this.setPrefHeight(height);
     }
+    
 
     private void toggleSettings() {
         if (this.getChildren().contains(settingsMenu)) {

--- a/src/main/java/org/openbase/bco/bcozy/view/CenterPane.java
+++ b/src/main/java/org/openbase/bco/bcozy/view/CenterPane.java
@@ -74,6 +74,7 @@ public class CenterPane extends StackPane {
         // Initializing components
         this.settingsMenu = loadSettingsMenu(height);
 
+        // Initializing view mode switch
         final FloatingPopUp viewModes = new FloatingPopUp(Pos.BOTTOM_RIGHT);
         viewModes.addParentElement(MaterialIcon.WEEKEND, () -> {
             appStateProperty.set(CenterPaneController.State.MOVEMENT);

--- a/src/main/java/org/openbase/bco/bcozy/view/FloatingPopUp.java
+++ b/src/main/java/org/openbase/bco/bcozy/view/FloatingPopUp.java
@@ -37,10 +37,6 @@ public class FloatingPopUp extends VBox {
         StackPane.setAlignment(this, position);
         this.translateYProperty().set(-Constants.INSETS);
     }
-    
-    public FloatingPopUp(final Pos position, final GlyphIcons... icons) {
-        this(position);
-    }
 
     public void addParentElement(final GlyphIcons icon, final Runnable handler) {
         this.addParentElement(icon, (event) -> handler.run());

--- a/src/main/java/org/openbase/bco/bcozy/view/FloatingPopUp.java
+++ b/src/main/java/org/openbase/bco/bcozy/view/FloatingPopUp.java
@@ -37,15 +37,9 @@ public class FloatingPopUp extends VBox {
         StackPane.setAlignment(this, position);
         this.translateYProperty().set(-Constants.INSETS);
     }
-
-    @Deprecated
-    public FloatingPopUp(GlyphIcons parent, GlyphIcons topChild, GlyphIcons bottomChild, Pos position) {
+    
+    public FloatingPopUp(final Pos position, final GlyphIcons... icons) {
         this(position);
-
-        addParentElement(parent, (Runnable) null);
-        addElement(topChild, (Runnable) null);
-        addElement(bottomChild, (Runnable) null);
-
     }
 
     public void addParentElement(final GlyphIcons icon, final Runnable handler) {

--- a/src/test/java/org/openbase/bco/bcozy/view/CenterPaneTest.java
+++ b/src/test/java/org/openbase/bco/bcozy/view/CenterPaneTest.java
@@ -1,19 +1,13 @@
 package org.openbase.bco.bcozy.view;
 
-import de.jensd.fx.glyphs.GlyphIcons;
 import de.jensd.fx.glyphs.materialdesignicons.MaterialDesignIcon;
 import de.jensd.fx.glyphs.materialicons.MaterialIcon;
 import javafx.application.Application;
 import javafx.geometry.Pos;
-import javafx.scene.Group;
 import javafx.scene.Scene;
 import javafx.scene.layout.BorderPane;
 import javafx.stage.Stage;
 import org.junit.Ignore;
-import org.junit.Test;
-import org.openbase.bco.bcozy.controller.CenterPaneController;
-
-import static org.junit.Assert.*;
 
 /**
  * @author vdasilva
@@ -42,7 +36,10 @@ public class CenterPaneTest extends Application {
         new CenterPaneController(centerPane);
         */
 
-        popUp = new FloatingPopUp(MaterialIcon.SETTINGS, MaterialDesignIcon.THERMOMETER_LINES, MaterialIcon.VISIBILITY, Pos.TOP_CENTER);
+        popUp = new FloatingPopUp(Pos.TOP_CENTER);
+        popUp.addParentElement(MaterialIcon.SETTINGS, this::setMaximizeAction);
+        popUp.addElement(MaterialDesignIcon.THERMOMETER_LINES, this::setMaximizeAction);
+        popUp.addElement(MaterialIcon.VISIBILITY, this::setMaximizeAction);
         popUp.addElement(MaterialIcon.ACCESS_ALARM, event -> System.out.println("ACCESS_ALARM"));
         pane.setLeft(popUp);
 

